templates/puppet.conf.erb

```diff
@@ -47,3 +47,13 @@
 
     syslogfacility = <%= @syslogfacility %>
 <% end -%>
+
+    
+<% if scope.lookupvar("puppet::server_directory_environments") -%>
+    environmentpath  = <%= scope.lookupvar("puppet::server_envs_dir") %>
+    basemodulepath   = <%= [scope.lookupvar("puppet::server_common_modules_path")].flatten.join(':') %>
+	<% elsif scope.lookupvar("puppet::server_git_repo") ||
+	        scope.lookupvar("puppet::server_dynamic_environments") -%>
+    manifest       = <%= scope.lookupvar("puppet::server_envs_dir") %>/$environment/manifests/site.pp
```

Comment:
I assume that "legacy way" is when we use `server_dynamic_environments` or `server_git_repo` with `manifest` directory. 
Anyway `manifest` should be in `[master] ` [section](https://docs.puppetlabs.com/puppet/latest/reference/dirs_manifest.html#with-puppet-master) and not `[main]` or `[server]`.
I will try to fix this too.